woocommerce/woocommerce-gutenberg-products-block
# Add subtotal to cart and checkout and update the CartLineItem component with new styles

As per the designs in this figma file: eaQcoMs7E8Ys0YgxfkOKxR-fi-0%3A1 this PR:

- Moves the quantity picker to below the product metadata in the cart block
- Removes the trash icon on mobile in the cart block and lets the "remove item" text be visible on mobile
- Adds a sale badge under the item total when the quantity is more than one and the item is on sale.
- Removes the link to the product on the checkout sidebar.
- Shows the individual item price below the product name in the checkout sidebar.
- Moves the "total" of each product (`individual total * quantity`) to its own "column" on the sidebar.

I have added some design changes here because adding the subtotal below the item name without making the other design changes caused this to look very messy.


Fixes #3629
Fixes #3631




#### Accessibility




### Screenshots

### Cart

#### Before
<img width="832" alt="Screenshot 2021-01-25 at 17 10 55" src="https://user-images.githubusercontent.com/5656702/105739704-4e338400-5f30-11eb-8481-ebf49fffe4d8.png">

#### After 
<img width="824" alt="Screenshot 2021-01-25 at 17 09 58" src="https://user-images.githubusercontent.com/5656702/105739595-2cd29800-5f30-11eb-99fe-ed629c6fa2d8.png">

### Checkout

#### Before
<img width="1114" alt="Screenshot 2021-01-26 at 11 16 40" src="https://user-images.githubusercontent.com/5656702/105838422-fc3e3d00-5fc7-11eb-94fc-f355b8f0544b.png">

#### After
<img width="1116" alt="Screenshot 2021-01-26 at 11 45 58" src="https://user-images.githubusercontent.com/5656702/105841161-1a0da100-5fcc-11eb-8755-f7f7ad22a97c.png">




### How to test the changes in this Pull Request:

1. Add items to your cart, some with sale prices, add multiple items etc.
2. See that the subtotal for each item is shown below the item's name, and the overall total is shown to the right. (Overall total is `item price * quantity`).
3. Try different themes out to see how the new design works there.

### Changelog

> Design tweaks to the cart page which move the quantity picker below each cart item and improve usability on mobile.
